ui/app/pages/settings/contact-list-tab/contact-list-tab.component.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import Button from '../../../components/ui/button'
+import Identicon from '../../../components/ui/identicon'
+import { addressSlicer } from '../../../helpers/utils/util'
+import { CONTACT_ADD_ROUTE, CONTACT_VIEW_ROUTE } from '../../../helpers/constants/routes'
+
+export default class ContactListTab extends Component {
+  static contextTypes = {
+    t: PropTypes.func,
+  }
+
+  static propTypes = {
+    addressBook: PropTypes.object,
+    history: PropTypes.object,
+    network: PropTypes.string,
+  }
+
+  renderAddresses () {
+    const { addressBook, history, network } = this.props
+    console.log('this network', network)
+
+    return (
+      <div>
+       { Object.keys(addressBook).filter(address => {
```

Comment:
I think the `Object.keys` and `addressBook[address]` might be unnecessary here due the changes made to the selector in `ui/app/pages/send/send.selectors.js`. `addressBook` should already be an array of objects.

Reply:
Actually, looking at line 34 suggests that previous comment is mistaken...

Reply:
Ah, it looks like the associated container file is importing `getAddressBook` from `selectors/selectors.js` but this PR changes `ui/app/pages/send/send.selectors.js`

Maybe we should change the method in `selectors/selectors.js`, and then code here can be simplified to:
```
addressBook.filter(entry => entry.chainId.toString() === network).map(({ address }) => ...`
```

Reply:
above is correct, yes: the state.metamask.addressBook changed from an array to an object